core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -185,23 +187,25 @@ object ChunkSetInfo extends StrictLogging {
   def setVectorPtr(infoPointer: NativePointer, colNo: Int, vector: BinaryVector.BinaryVectorPtr): Unit =
     UnsafeUtils.setLong(infoPointer + OffsetVectors + 8 * colNo, vector)
 
-  // deprecated - Use reader api instead TODO remove this API
-  def getNumRows(infoBytes: Array[Byte]): Int = UnsafeUtils.getInt(infoBytes, UnsafeUtils.arayOffset + OffsetNumRows)
+  def getNumRows(info: ByteBuffer): Int = getNumRows(fromByteBuffer(info), 0)
+
+  def getChunkID(info: ByteBuffer): ChunkID = getChunkID(fromByteBuffer(info), 0)
+
+  def getIngestionTime(info: ByteBuffer): Long = getIngestionTime(fromByteBuffer(info), 0)
+
+  def getStartTime(info: ByteBuffer): Long = startTimeFromChunkID(getChunkID(info))
+
+  def getEndTime(info: ByteBuffer): Long = getEndTime(fromByteBuffer(info), 0)
+
+  def getNumRows(info: Array[Byte]): Int = getNumRows(fromArray(info), 0)
```

Comment:
We have to double check impact of object creation here. This was the reason I did not remove the original API in my on-heap vectors PR.

Reply:
The code here is highly inlinable. The wrapper objects would just turn into stack-allocated local variables.